src/phoenix/core/model_schema.py

```diff
@@ -179,10 +181,26 @@ class DatasetRole(IntEnum):
 REFERENCE = DatasetRole.REFERENCE
 
 
-@lru_cache(maxsize=len(DatasetRole))
-def _series_nan(length: int) -> "pd.Series[float]":
-    """Useful as a substitute for a non-existent column."""
-    return pd.Series(np.full(length, float("nan")))
+@dataclass(frozen=True, repr=False, eq=False)
+class _ConstantValueSeriesFactory:
```

Comment:
agree that "privatizing" is generally no worth it, but alternatively, we can also see the `_` prefix as meaning `local` instead of `private`, and the `_` works quite well as a visual aid for that purpose